### ISSUE TITLE
Control flow stuck in 2.48.0

### ISSUE THREAD

**Issue by hankduan** [SeleniumHQ/selenium#1207]
The following scenario gets stuck in 2.48.0, but not 2.47.0.

Repo case:

```javascript
var webdriver = require('selenium-webdriver');
var driver = new webdriver.Builder().
  usingServer('http://localhost:4444/wd/hub').
  withCapabilities(webdriver.Capabilities.chrome()).
  build();

driver.get('http://www.google.com');

var getRandomText = function() {
  return driver.findElement(webdriver.By.id('prm')).getText();
}
var textPromise1 = getRandomText();
var textPromise2 = getRandomText();
webdriver.promise.when(textPromise1).then(function(actual) {
  return webdriver.promise.when(textPromise2).then(function(expected) {
    console.log('cannot get here');
  });
});
```

**Comment by jleyba** [comment]
Reduced to

```
var webdriver = require('selenium-webdriver');

var flow = webdriver.promise.controlFlow();

var p1 = flow.execute(() => console.log('in p1'), 'execute p1');
var p2 = flow.execute(() => console.log('in p2'), 'execute p2');

p1.then(() => { console.log('post p1'); return p2; })
  .then(() => { console.log('post p1 & p2') });
```

**Comment by jleyba** [comment]
Published 2.48.1 to npm
